src/main/java/com/example/onlinebookshop/entities/Book.java

```diff
@@ -0,0 +1,97 @@
+package com.example.onlinebookshop.entities;
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.validation.constraints.NotNull;
+import java.math.BigDecimal;
+
+@Entity
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull(message = "Title may not be null")
+    private String title;
+    @NotNull(message = "Author may not be null")
+    private String author;
+    @NotNull(message = "isbn may not be null")
+    @Column(unique = true)
+    private String isbn;
+    @NotNull(message = "Price may not be null")
```

Comment:
If you want to enforce the non-null constraint at the database level, it is recommended to use the @Column(nullable = false) annotation. This annotation instructs JPA to create the corresponding column in the database table with a NOT NULL constraint.